synapse/storage/databases/main/roommember.py

```diff
@@ -419,7 +419,11 @@ async def get_rooms_for_local_user_where_membership_is(
         )
 
         # Now we filter out forgotten and excluded rooms
-        rooms_to_exclude = await self.get_forgotten_rooms_for_user(user_id)
+        rooms_to_exclude: AbstractSet[str] = set()
+
+        # Users can't forget joined/invited rooms, so we skip the check for such look ups.
+        if not all(m in (Membership.JOIN, Membership.INVITE) for m in membership_list):
```

Comment:
I wonder if this is clearer
```suggestion
        if any(m not in (Membership.JOIN, Membership.INVITE) for m in membership_list):
```

Does this optimisation apply often? I'm finding it hard to imagine that many users have not left a single room.